src/compas-ui/helpers/compas-scl-list.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  css,
+  customElement,
+  html,
+  LitElement,
+  property,
+  PropertyValues,
+  state,
+  TemplateResult,
+} from 'lit-element';
+
+import '@material/mwc-icon';
+import '@material/mwc-list';
+import '@material/mwc-list/mwc-list-item';
+import { translate } from 'lit-translate';
+
+import { SelectedItemsChangedEvent } from '../../to-be-made-a-dependency/oscd-filter-button.js';
+import '../../to-be-made-a-dependency/filtered-list.js';
+import '../../to-be-made-a-dependency/oscd-filter-button.js';
+import './compas-loading.js';
+
+/* Event that will be used when a SCL is selected from a list of SCL Documents. */
+export interface SclSelectedDetail {
+  docId: string;
+}
+export type SclSelectedEvent = CustomEvent<SclSelectedDetail>;
+export function newSclSelectedEvent(docId: string): SclSelectedEvent {
+  return new CustomEvent<SclSelectedDetail>('scl-selected', {
+    bubbles: true,
+    composed: true,
+    detail: { docId },
+  });
+}
+
+@customElement('compas-scl-list')
+export class CompasSclList extends LitElement {
+  @property()
+  type?: string;
+  @property()
+  nameSpace = '';
+
+  @property()
+  private items: Element[] | undefined;
+
+  @property()
+  private labels: string[] = [];
+
+  @property()
+  private selectedLabels: string[] = [];
+
+  @property()
+  private get filteredItems(): Element[] | undefined {
+    // If items are still being retrieved, return undefined.
+    if (!this.items) {
+      return undefined;
+    }
+
+    // If all labels are selected, show all items, including the ones not having a Label.
+    if (this.labels.length === this.selectedLabels.length) {
+      return this.items;
+    }
+
+    return this.items.filter(item => {
+      const labels = Array.from(item.querySelectorAll('Label') ?? [])
+        .map(element => element.textContent)
+        .filter(value => !!value) as string[];
+      return (
+        labels.filter(label => this.selectedLabels.includes(label)).length > 0
+      );
+    });
+  }
+
+  render(): TemplateResult {
+    if (!this.items) {
+      return html`
```

Comment:
maybe move this if-statement implementation to a different function?

something like this maybe?:

```ts

private renderLoadingBar(): TemplateResult {
  return html`
    <compas-loading .message=${translate('compas.loading.files')}>
   </compas-loading>
  `;
}

private renderNoScls(): TemplateResult {
  return html`<mwc-list>
    <mwc-list-item><i>${translate('compas.noScls')}</i></mwc-list-item>
  </mwc-list>`;
}

private renderSclList(): TemplateResult {
  return html`
  <div class="filters">
    ...
  </div>`;
}

render(): TemplateResult {
  return !this.items ? this.renderLoadingBar() : this.items.length <= 0 ? this.renderNoScls() : this. renderSclList();
}

```

This resolves in a render function that has a single return statement
https://www.tomdalling.com/blog/coding-tips/coding-tip-have-a-single-exit-point/

Reply:
☑️